client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamClientImpl.java

```diff
@@ -48,7 +51,11 @@
 
     @Override
     public ByteStreamReader createByteStreamReader(String streamName) {
-        return createByteStreamReaders(new Segment(scope, streamName, 0));
+        // Fetch the segments pointing to the current HEAD of the stream.
+        Map<Segment, Long> segments = Futures.getThrowingException(controller.getSegmentsAtTime(Stream.of(scope, streamName), 0L));
+        Preconditions.checkState(segments.size() == 1, "Stream is configured with more than one segment or has none");
```

Comment:
For debug-ability it would be good to have two different errors or include the size in the error message.

Reply:
improved it.